models/collector_base.go

```diff
@@ -142,8 +142,8 @@ type CollectorBase struct {
 	// Read Only: true
 	LastSentNotificationOnLocal string `json:"lastSentNotificationOnLocal,omitempty"`
 
-	// Whether to create a collector device when instance collector
-	NeedAutoCreateCollectorDevice bool `json:"needAutoCreateCollectorDevice,omitempty"`
+	// Whether to create a collector device when instance collector, the default value is true
+	NeedAutoCreateCollectorDevice interface{} `json:"needAutoCreateCollectorDevice,omitempty"`
```

Comment:
@JeremyTangCD Why are these properties interfaces now instead of bool?

Reply:
@woz5999 
You can get the details in DEV-43101.
Same reason as https://github.com/logicmonitor/lm-sdk-go/pull/6, the default value in santaba is true, if we set the value as false, go sdk will treat this value as the default value and does not send it to santaba, then we get in santaba will be true
In https://github.com/logicmonitor/lm-sdk-go/pull/6, we just delete the "omitempty" tag from the sdk manually.
But since the sdk should only be auto-generated, we need to improve this case by improve the annotation in santaba. We could not change this field to be required, so we only have to change the type of this value and add some description for it.